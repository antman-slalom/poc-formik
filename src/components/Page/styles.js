const styles = theme => ({
    ...theme.globalClasses,
    ErrorMessage: {
        color: theme.globalClasses.errorPrimary.color
    },
    Page: {
        padding: '1.875rem',
    },
    PageComponent: {
        padding: '1.25rem 0rem',
    }
});

export default styles;
