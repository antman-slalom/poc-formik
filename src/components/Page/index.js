import React from 'react';

import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/styles';

import styles from './styles';

function validationError(showError, typoStyle) {
    if (showError) {
        return (
            <Typography className={ typoStyle }>Please check for validation errors</Typography>
        );
    }
    return '';
}

function saveButton(saveButtonClick, saveButtonDisabled, saveButtonVisible) {
    if (saveButtonVisible) {
        return (
            <Button
                color='primary'
                disabled={ saveButtonDisabled }
                id='SaveButton'
                onClick={ saveButtonClick }
                size='small'
                variant='contained'
            >
                SAVE
            </Button>
        );
    }
    return '';
}

class Page extends React.Component {
    render() {
        const {
            children, classes, title, subTitle
        } = this.props;

        return (
            <Box className={ classes.Page }>
                <Box className={ classes.PageComponent } id='title'>
                    <Grid container>
                        <Grid item xs={ 12 }>
                            { (subTitle) && <Typography variant='h5'>{ subTitle }</Typography> }
                            <Typography variant='h3'>{ title }</Typography>
                        </Grid>
                    </Grid>
                </Box>
                <Box className={ classes.PageComponent } id='content'>
                    { children }
                </Box>
            </Box>
        );
    }
}

export default withStyles(styles)(Page);
