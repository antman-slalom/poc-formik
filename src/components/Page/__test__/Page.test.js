import React from 'react';

import { shallow } from 'enzyme';
import { unwrap } from '@material-ui/core/test-utils';

import Page from '../index';

describe('Page', () => {
    let Unwrapped;
    let wrapper;

    const props = {
        classes: {},
        title: 'title',
        subTitle: 'user'
    };

    beforeEach(() => {
        Unwrapped = unwrap(Page);
        wrapper = shallow(<Unwrapped { ...props } />);
    });

    it('matches snapshot', () => {
        expect(wrapper.instance()).toMatchSnapshot();
    });

    it('should render', () => {
        expect(wrapper.instance()).not.toBeNull();
    });

    // TODO - fix tests
    it('should render with children', () => {
        // const children = ['<div>Child 1</div>', '<div>Child 2</div>'];
        // const shallowWrapper = shallow(<Unwrapped { ...props } >{ children }</Unwrapped>);

        // expect(shallowWrapper.find(Paper).children()).toHaveLength(2);
    });
});
