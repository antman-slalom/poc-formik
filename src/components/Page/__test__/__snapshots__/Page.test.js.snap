// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page matches snapshot 1`] = `
Page {
  "context": Object {},
  "props": Object {
    "classes": Object {},
    "saveButtonDisabled": false,
    "saveButtonVisible": false,
    "showPageValidationError": false,
    "subTitle": "user",
    "title": "title",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
      },
      "_element": <Page
        classes={Object {}}
        saveButtonDisabled={false}
        saveButtonVisible={false}
        showPageValidationError={false}
        subTitle="user"
        title="title"
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <ForwardRef(StyledComponent)>
        <ForwardRef(StyledComponent)
          id="title"
        >
          <ForwardRef(WithStyles)
            container={true}
          >
            <ForwardRef(WithStyles)
              item={true}
              xs={9}
            >
              <ForwardRef(WithStyles)
                variant="h5"
              >
                user
              </ForwardRef(WithStyles)>
              <ForwardRef(WithStyles)
                variant="h3"
              >
                title
              </ForwardRef(WithStyles)>
            </ForwardRef(WithStyles)>
            <ForwardRef(WithStyles)
              item={true}
              xs={2}
            >
              
            </ForwardRef(WithStyles)>
            <ForwardRef(WithStyles)
              item={true}
              xs={1}
            >
              
            </ForwardRef(WithStyles)>
          </ForwardRef(WithStyles)>
        </ForwardRef(StyledComponent)>
        <ForwardRef(StyledComponent)
          id="content"
        />
      </ForwardRef(StyledComponent)>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
